### ISSUE TITLE
`help` should try getting help for external commands

### ISSUE THREAD

**Issue by kaathewisegit** [nushell/nushell#11669]
Currently `help` only works on Nushell commands.  I think it'd be a good idea to try a couple of heuristics on external commands, something akin to:

1. Try to fetch info about Nushell built-in stuff.
2. Try calling `man <command>`.
3. If that fails, try to call `<command> help`, `command --help`, and `command -h`.
4. And if none of the above work, give up.

**Comment by fdncred** [comment]
There is a command_not_found hook. I wonder if that could be used?

**Comment by weirdan** [comment]
Point 3 is dangerous as there's no guarantee how the command would interpret those arguments. For example, `du -h` does *not* output help. 

Point 2 is already done.

**Comment by KAAtheWiseGit** [comment]
Hmm, calling man pages doesn't work for me on 0.96:

```Nushell
~> man git  # works
~> help git
Error: nu::parser::not_found

  × Not found.
   ╭─[entry #9:1:6]
 1 │ help git
   ·      ─┬─
   ·       ╰── did not find anything under this name
   ╰────
```

The point about calling commands being dangerous is absolutely correct, I haven't thought about it at all

**Comment by weirdan** [comment]
Ah, I see. `std help` does have a fallback to `man`, while core help doesn't seem to have it.

You can try adding `use std help` somewhere in your `config.nu` and see if it works for you. The output is a bit different compared to the standard help viewer.

**Comment by weirdan** [comment]
@fdncred I guess the question now is whether the current `std help` will be the base for the new help system once the new parser is ready. If it is, then this issue is already addressed as far as it could (safely) be.

**Comment by fdncred** [comment]
@weirdan I think that's our goal. Move help to stdlib, put it in the prelude, remove all the rust help code. It all really depends on what type of a hit it is on startup with the new-nu-parser.
